### ISSUE TITLE
Tests fail if ran in random order

### ISSUE THREAD

**Issue by cphyc** [pynbody/pynbody#746]
**Describe the bug**
When running the test suite, some tests fail when run in a given order.

**To Reproduce**

Install pytest-random-order (`pip install pytest-random-order`) then run the tests:
```bash
# This succeeds
pytest --random-order --random-order-seed=509902 tests/nchilada_test.py -v
# This fails
pytest --random-order --random-order-seed=1234 tests/nchilada_test.py -v
```

Output:
```
================================================================================================ test session starts ================================================================================================
platform linux -- Python 3.12.0, pytest-7.4.3, pluggy-1.3.0 -- REDACTED/py312/bin/python3.12
cachedir: .pytest_cache
Using --random-order-bucket=module
Using --random-order-seed=1234

rootdir: /home/cphyc/Documents/prog/pynbody
configfile: pyproject.toml
plugins: random-order-1.1.0
collected 5 items                                                                                                                                                                                                   

tests/nchilada_test.py::test_get PASSED                                                                                                                                                                       [ 20%]
tests/nchilada_test.py::test_HI_not_present_for_non_gas_particles PASSED                                                                                                                                      [ 40%]
tests/nchilada_test.py::test_partial_loading PASSED                                                                                                                                                           [ 60%]
tests/nchilada_test.py::test_array_completion_unit_sanity PASSED                                                                                                                                              [ 80%]
tests/nchilada_test.py::test_get_gas PASSED                                                                                                                                                                   [100%]

================================================================================================= 5 passed in 0.88s =================================================================================================


================================================================================================ test session starts ================================================================================================
platform linux -- Python 3.12.0, pytest-7.4.3, pluggy-1.3.0 -- /home/cphyc/miniconda3/envs/py312/bin/python3.12
cachedir: .pytest_cache
Using --random-order-bucket=module
Using --random-order-seed=509902

rootdir: /home/cphyc/Documents/prog/pynbody
configfile: pyproject.toml
plugins: random-order-1.1.0
collected 5 items                                                                                                                                                                                                   

tests/nchilada_test.py::test_array_completion_unit_sanity FAILED                                                                                                                                              [ 20%]
tests/nchilada_test.py::test_HI_not_present_for_non_gas_particles PASSED                                                                                                                                      [ 40%]
tests/nchilada_test.py::test_partial_loading PASSED                                                                                                                                                           [ 60%]
tests/nchilada_test.py::test_get PASSED                                                                                                                                                                       [ 80%]
tests/nchilada_test.py::test_get_gas PASSED                                                                                                                                                                   [100%]

===================================================================================================== FAILURES ======================================================================================================
_________________________________________________________________________________________ test_array_completion_unit_sanity _________________________________________________________________________________________

    def test_array_completion_unit_sanity():
>       del f['pos']

tests/nchilada_test.py:194: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <SimSnap "testdata/nchilada_test/12M.00001" len=233217>, name = 'pos'

    def __delitem__(self, name):
        if name in self._family_arrays:
            # mustn't have simulation-level array of this name
            assert name not in self._arrays
            del self._family_arrays[name]
    
            for v in self._family_derived_array_names.values():
                if name in v:
                    del v[v.index(name)]
    
        else:
>           del self._arrays[name]
E           KeyError: 'pos'

pynbody/snapshot/__init__.py:326: KeyError
============================================================================================== short test summary info ==============================================================================================
FAILED tests/nchilada_test.py::test_array_completion_unit_sanity - KeyError: 'pos'
============================================================================================ 1 failed, 4 passed in 0.91s ============================================================================================
```

**Expected behaviour or result**
The tests should be sufficiently compartmentalized to work regardless of their order.

**Additional context**
This causes issues while debugging #745.

**Comment by apontzen** [comment]
For reference I think it's seed 509902 that fails and 1234 succeeds (the opposite of what's written above)

**Comment by apontzen** [comment]
This should be fixed in #747 - please could you check?

**Comment by apontzen** [comment]
I think this issue is resolved, feel free to re-open if I'm wrong!
